### ISSUE TITLE
Settings for the Monospace Font will be forgotten when DBeaver is closed

### ISSUE THREAD

**Issue by DudeMcGee** [dbeaver/dbeaver#35872]
### Description

When I change the Monospace Font settings in the preferences the changes only apply during the current DBeaver session. When I close DBeaver and start it again, the settings are still shown in the preferences, but the used font is the default one and not the one written into the settings.

### DBeaver Version

CE 24.2.2.202410061621

### Operating System

Kali Linux (current, rolling release)

### Database and driver

Mysql/MariaDB, Postgresql

### Steps to reproduce

Go into settings.
Change Monospace font.
See the change, e.g. in a separate value field, showing the contents of a TEXT column.
Close DBeaver and open it again.
The appearance is the same as before the font change.

**Comment by E1izabeth** [comment]
I tried to reproduce the issue, and it works fine for me.
You're pressing the `Apply and Close` or `Apply` button in the Preferences dialog, right?

**Comment by DudeMcGee** [comment]
Without using the Apply buttons you won't see any change at no time, right? So, yes, I used these two buttons in the possible different ways to achieve a permanent change in appearance, but it didn't work.
In order to avoid any possible misunderstanding I have included three screenshots:
Picture a shows a fresh DBeaver session.
Picture b shows the same session after explicitly having changed the Monospace font. So, this picture shows how things should permanently look like after changing the font.
Picture c shows DBeaver after being shut down and relaunched. You see, that the appearance is the same as if never any change of the Monospace font had taken place.
Thanks for taking care and I hope you can somehow reproduce the error or find colleagues, mates, whatever who can confirm the problem. It's simply very unlikely that I'm the only one who experiences this issue. 
Thank you!
![a](https://github.com/user-attachments/assets/b9d8a2fb-a013-44f8-8526-c56c986de783)
![b](https://github.com/user-attachments/assets/91c52311-e070-4470-9b0b-46c77a82a88d)
![c](https://github.com/user-attachments/assets/e89e4046-38cc-4732-83b0-b0ffedddd7ef)

**Comment by E1izabeth** [comment]
I see, thank you for quick answer. Maybe it's a linux-specific issue. We'll try reproducing again

**Comment by DudeMcGee** [comment]
Thank you for your support and your efforts to help! I'll be glad to help out with further tests and experiments if needed!

**Comment by Matvey16** [comment]
@DudeMcGee It didn't reproduce for me on Ubuntu, so we would need to find the difference. First of all how did you install DBeaver? And I would like to know how exactly did you change the font with details, maybe I do something differently

**Comment by DudeMcGee** [comment]
Yes, I can only reproduce the problem on one of my Kali Linux computers. The problem is not present on a second Kali machine and also not on a Linux Mint computer.
The installation on all machines was a two-step procedure:

1. Download the DBeaver-CE
2. dpkg -i dbeaver... (as root or using sudo)

That's all, no other preparations before use.

I change the font as follows:

1. Window
2. Preferences
4. On the left tree panel to  User interface, Appearance, Colors and fonts
5. On the right hand area double-click on DBeaver fonts
6. Double-click Monospace font
7. Choose another font than the one which is currently selected
8. Press Apply
9. Press Apply and Close

That's the process. After a restart the change is lost.

There's no sense in testing under Ubuntu. To me it is an open question why one Kali machine shows this error permanently in 100% of all cases, whereas the other one never has this problem. It could play a role how the desktop is organized (system-wide theme, XFCE configuration). I'll try to find out more.

**Comment by DudeMcGee** [comment]
Additional information on the machine which shows the error:

OS: Kali GNU/Linux Rolling x86_64 
Host: ZenBook UX425QA_UM425QA 1.0 
Kernel: 6.11.2-amd64 
Uptime: 9 hours, 14 mins 
Packages: 4845 (dpkg), 10 (flatpak) 
Shell: zsh 5.9 
Resolution: 1920x1080 
DE: Xfce 4.18 
WM: Xfwm4 
Theme: Kali-Dark [GTK2], adw-gtk3-dark [GTK3] 
Icons: Flat-Remix-Blue-Dark [GTK2/3] 
Terminal: konsole 
CPU: AMD Ryzen 5 5600H with Radeon Graphics (12) @ 4.280GHz 
GPU: AMD ATI Radeon Vega Series / Radeon Vega Mobile Series 
Memory: 6848MiB / 15393MiB

**Comment by DudeMcGee** [comment]
Unfortunately, the Kali computer on which this error does not occur, is not available at the moment, but information on it can be given in about two or three weeks when I will have access to it again.

**Comment by DudeMcGee** [comment]
In the meantime, everyone of my three computers which run DBeaver shows this error, even with the latest update at the time of this writing. Two of the computers run Kali Linux, the other one runs Linux Mint.  

In addition to not restoring the user-chosen font, the value panel also does not remember the setting of the word-wrap toggle switch once the font error has come up.

I also found out, that this font error does occur on Windows machines, too. So that's not a Linux problem. And in addition, there are older reports on this issue page here which document that this error is relatively old and goes back to 2023 at least.

One remark: It is not helpful when the support points out that they can't reproduce the problem. This issue has been experienced and documented by users, and since I'm not a professional programmer I cannot tell you where the cause of the bug is hidden. But when I experience this on three different machines over the course of at least three different DBeaver releases, now running the latest OS update plus the latest DBeaver edition and the problem still there, you should acknowledge that there is a problem. 

I have done hours-long analyses of the problem and tried at least half a dozen remove and re-install circles under different conditions to no avail. 

My workaround is now: Set the main font to the font setting you wish to have for the value panel. That's not quite ideal but at least feasible.

**Comment by E1izabeth** [comment]
Please send us log files. You can collect them to archive automatically with `Help->Collect Diagnostic Info`
![image](https://github.com/user-attachments/assets/51c7ca2e-6be2-489e-b495-028b9aa0e386)

**Comment by DudeMcGee** [comment]
I have performed the following steps:

1. In my open DBeaver session I have written the log file starting with A_ in its name.
2. I then have changed the monofont from Ubuntu Sans Mono 12 Bold to Ubuntu Sans Mono 16 Bold. This change could be seen in the value panel, all quite nice up to this point.
3. I quit the session.
4. I re-started DBeaver. I selected a column and saw in the value panel that the font was not the most recent one, my change had gone lost somehow.
5. I wrote a second log file, starting with B_.

I really do hope that you find a trace of the wrong behaviour in these data! And if I can be of any assistance, I'd be glad to help you finding the bug!

[A_dbeaver-diagnostic-info-1736965722754.zip](https://github.com/user-attachments/files/18428846/A_dbeaver-diagnostic-info-1736965722754.zip)
[B_dbeaver-diagnostic-info-1736965938239.zip](https://github.com/user-attachments/files/18428847/B_dbeaver-diagnostic-info-1736965938239.zip)

**Comment by DudeMcGee** [comment]
BTW: Some bugs more or less, I wish to thank all of you developers, assistants and supporters for DBeaver! I'm using the community edition just for private purposes, fun and learning, and I'm glad that there is such a good product to work with for free! So, I hope we can eliminate this silly bug and have a good time! Thanks!

**Comment by E1izabeth** [comment]
Thank you

**Comment by arhayka** [comment]
Thank you for the report. We still can't reproduce this.
Now, this is reproducible on Kali Linux only.

**Comment by muchosnachos** [comment]
In support of the need to repro this issue, I can add my experience.  I'm on Max OS, and I can repro this issue:

Mac OS: 26.0.1
DBeaver 25.2.1.202509211659

Steps to repro (to support my low-vision condition):
1. Windows -> Preferences
2. User Interface > Appearance > Colors and Fonts >> Beaver Fonts:
3a. Main font, click <Edit>, Change Font to 18, Close windows, Apply (immediately see larger fonts in Preferences pane)
3b. Monospace font, click <Edit>, Change Font to 18, Close windows, Apply (and will see the results in the SQL Editor in a moment when Prefences closes)
4. Click Apply and Close (all visible fonts are larger now (18 point), this is good throughout the session
5. Close DBeaver, and re-start.  The visual settings show small fonts (13 point) again, but if I re-open preferences... it display 18 point.  Clicking Apply or Apply and Close does NOT apply the 18 point size.  I need to reset the fonts, Apply and Close, then open the preferences again, set to 18 and the settings will work again for the session.

Notes: 
- I'm using Dark Theme, so I tried turning "Enable theming" off.  Re-starting still lost the user preferences, though the setting still displayed 18 even though fonts were clearly on 13 point.
- I was also using ht e"Color and font theme: Classic... so I turn it back do Default AND it seems like for now my settings persist re-starts!
I wonder if the combination of Classic Color and font theme, and these font size preference have a conflict somewhere.  I'll try out the Default Color and font theme for a while, with Dark Theme selected to see if this holds up permanently.

Hope this helps... as this was challenging to use the workaround each restart.

**Comment by DudeMcGee** [comment]
Thank you for your report, muchosnachos, and thanks for the work you've invested into the issue!

As I wrote on January 15, 2025: "My workaround is now: Set the main font to the font setting you wish to have for the value panel. That's not quite ideal but at least feasible."

Maybe this helps you out, it's quite easy to use this method although it takes away flexibility in designing your GUI experience.

Although there's clear evidence for the bug (several people have reported the problem over a long time) the developer team seams to be unable to even locate the problem in their code. That appears very strange to me, one could think the main developer is on a journey around the world or something. Don't know. 

I have a feeling that dBeaver won't get rid of the problem within this decade.
